### PR TITLE
Add recipe for threadpoolctl

### PR DESCRIPTION
**Pull request by jeremiedbb**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jeremiedbb** [comment]
@conda-forge/staged-recipes I think this is ready.

The appveyor error seems unrelated:
```py
Build started
git clone -q https://github.com/conda-forge/staged-recipes.git C:\projects\staged-recipes
git fetch -q origin +refs/pull/10808/merge:
git checkout -qf FETCH_HEAD
Running Install scripts
powershell -Command "(New-Object Net.WebClient).DownloadFile('https://raw.githubusercontent.com/conda-forge/conda-forge-build-setup-feedstock/master/recipe/ff_ci_pr_build.py', 'ff_ci_pr_build.py')"
ff_ci_pr_build -v --ci "appveyor" "%APPVEYOR_ACCOUNT_NAME%/%APPVEYOR_PROJECT_SLUG%" "%APPVEYOR_BUILD_NUMBER%" "%APPVEYOR_PULL_REQUEST_NUMBER%"
Checking to see if this PR build is outdated.
Traceback (most recent call last):
  File "C:\projects\staged-recipes\ff_ci_pr_build.py", line 201, in <module>
    sys.exit(main())
  File "C:\projects\staged-recipes\ff_ci_pr_build.py", line 192, in main
    exit_code = int(appveyor_check_latest_pr_build(repo, pr, bld) is False)
  File "C:\projects\staged-recipes\ff_ci_pr_build.py", line 117, in appveyor_check_latest_pr_build
    data = request_json(url.format(repo=repo, total_builds=total_builds), headers=headers)
  File "C:\projects\staged-recipes\ff_ci_pr_build.py", line 46, in request_json
    with contextlib.closing(urlopen(request)) as response:
  File "C:\Python27\lib\urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "C:\Python27\lib\urllib2.py", line 429, in open
    response = self._open(req, data)
  File "C:\Python27\lib\urllib2.py", line 447, in _open
    '_open', req)
  File "C:\Python27\lib\urllib2.py", line 407, in _call_chain
    result = func(*args)
  File "C:\Python27\lib\urllib2.py", line 1241, in https_open
    context=self._context)
  File "C:\Python27\lib\urllib2.py", line 1198, in do_open
    raise URLError(err)
urllib2.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727)>
Command exited with code 1
```

**Comment by jeremiedbb** [comment]
@conda-forge-admin, please ping team

**Comment by jeremiedbb** [comment]
I think this is ready.

The appveyor error seems unrelated:

Build started
git clone -q https://github.com/conda-forge/staged-recipes.git C:\projects\staged-recipes
git fetch -q origin +refs/pull/10808/merge:
git checkout -qf FETCH_HEAD
Running Install scripts
powershell -Command "(New-Object Net.WebClient).DownloadFile('https://raw.githubusercontent.com/conda-forge/conda-forge-build-setup-feedstock/master/recipe/ff_ci_pr_build.py', 'ff_ci_pr_build.py')"
ff_ci_pr_build -v --ci "appveyor" "%APPVEYOR_ACCOUNT_NAME%/%APPVEYOR_PROJECT_SLUG%" "%APPVEYOR_BUILD_NUMBER%" "%APPVEYOR_PULL_REQUEST_NUMBER%"
Checking to see if this PR build is outdated.
Traceback (most recent call last):
  File "C:\projects\staged-recipes\ff_ci_pr_build.py", line 201, in <module>
    sys.exit(main())
  File "C:\projects\staged-recipes\ff_ci_pr_build.py", line 192, in main
    exit_code = int(appveyor_check_latest_pr_build(repo, pr, bld) is False)
  File "C:\projects\staged-recipes\ff_ci_pr_build.py", line 117, in appveyor_check_latest_pr_build
    data = request_json(url.format(repo=repo, total_builds=total_builds), headers=headers)
  File "C:\projects\staged-recipes\ff_ci_pr_build.py", line 46, in request_json
    with contextlib.closing(urlopen(request)) as response:
  File "C:\Python27\lib\urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "C:\Python27\lib\urllib2.py", line 429, in open
    response = self._open(req, data)
  File "C:\Python27\lib\urllib2.py", line 447, in _open
    '_open', req)
  File "C:\Python27\lib\urllib2.py", line 407, in _call_chain
    result = func(*args)
  File "C:\Python27\lib\urllib2.py", line 1241, in https_open
    context=self._context)
  File "C:\Python27\lib\urllib2.py", line 1198, in do_open
    raise URLError(err)
urllib2.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727)>
Command exited with code 1

**Comment by jeremiedbb** [comment]
thanks @isuruf. Is there something else to be done ?

File: recipes/threadpoolctl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    Python helpers to introspect and limit the number of threads used
+    in native libraries that handle their own internal threadpool
+    (BLAS and OpenMP implementations).
+  doc_url: https://github.com/joblib/threadpoolctl
+  dev_url: https://github.com/joblib/threadpoolctl
+
+extra:
+  recipe-maintainers:
+    - ogrisel
+    - tomMoral
```
Comment (scopatz) [inline recipes/threadpoolctl/meta.yaml]: Are you willing to be maintainers:
@ogrisel
@tomMoral

Comment (jeremiedbb) [reply recipes/threadpoolctl/meta.yaml]: don't know if it's enough but they responded positively in a draft I made before submitting this PR: https://github.com/jeremiedbb/staged-recipes/pull/1

Comment (tomMoral) [reply recipes/threadpoolctl/meta.yaml]: I am!

Comment (ogrisel) [reply recipes/threadpoolctl/meta.yaml]: Yes!

File: recipes/threadpoolctl/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  imports:
+    - threadpoolctl
+  commands:
+    - python -c "import numpy; from threadpoolctl import threadpool_info; threadpool_info()"
+    - python -c "import numpy; from threadpoolctl import threadpool_limits; threadpool_limits(1)"
+    - python -m threadpoolctl -i numpy
+
+about:
+  home: https://github.com/joblib/threadpoolctl
+  license: BSD
```
Comment (scopatz) [inline recipes/threadpoolctl/meta.yaml]: Which version of BSD is this?

Comment (jakirkham) [reply recipes/threadpoolctl/meta.yaml]: Looks like [BSD-3-Clause]( https://github.com/joblib/threadpoolctl/blob/master/LICENSE ). Can you please confirm/deny @jeremiedbb and if so commit this (or other appropriate) change? 🙂

```suggestion
  license: BSD-3-Clause
```

File: recipes/threadpoolctl/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  run:
+    - python >=3.5
+
+test:
+  requires:
+    - numpy
+  imports:
+    - threadpoolctl
+  commands:
+    - python -c "import numpy; from threadpoolctl import threadpool_info; threadpool_info()"
```
Comment (isuruf) [inline recipes/threadpoolctl/meta.yaml]: Can you print the info as well?

Comment (jakirkham) [reply recipes/threadpoolctl/meta.yaml]: Should this just be moved to `run_test.py`? Would make it a bit easier to follow 🙂

File: recipes/threadpoolctl/meta.yaml (hunk 2 above)
Comment (jeremiedbb) [reply recipes/threadpoolctl/meta.yaml]: it is. I updated.

File: recipes/threadpoolctl/meta.yaml (hunk 3 above)
Comment (jeremiedbb) [reply recipes/threadpoolctl/meta.yaml]: I added a `run_test.py`

**Comment by jakirkham** [comment]
LGTM. Thanks @jeremiedbb! 😀

Any other comments @isuruf? 😉

**Comment by jeremiedbb** [comment]
@isuruf, @scopatz, is there something holding this ?

**Comment by jeremiedbb** [comment]
Thanks !

**Comment by jakirkham** [comment]
Thanks @jeremiedbb! 😄

**Pull request merged**